doc/vxlan/Overlay ECMP ehancements.md

```diff
@@ -0,0 +1,227 @@
+ - adv_prefix
+
+```
+
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}}  
+    “endpoint”: {{ip_address1},{ip_address2},...} 
+    “endpoint_monitor”: {{ip_address1},{ip_address2},...} (OPTIONAL) 
+    “mac_address”: {{mac_address1},{mac_address2},...} (OPTIONAL) 
+    “monitoring”: {{“custom”}} (OPTIONAL)                                  <<<< New Field
+    “vni”: {{vni1},{vni2},...} (OPTIONAL) 
+    “weight”: {{w1},{w2},...} (OPTIONAL) 
+    “profile”: {{profile_name}} (OPTIONAL) 
+    “primary”: {{ip_address1}, {ip_address2}} (OPTIONAL)                   <<<< New Field    
+    “profile”: {{profile_name}} (OPTIONAL)  
+    “adv_prefix”: {{prefix}} (OPTIONAL)                                    <<<< New Field
+    “rx_monitor_timer”: {time in milliseconds} (OPTIONAL)                  <<<< New Field
+    “tx_monitor_timer”: {time in milliseconds} (OPTIONAL)                  <<<< New Field
+    “check_directly_connected”: {{true|false}} (OPTIONAL)                  <<<< New Field
+```
+
+
+```
+primary                  = ipv4/v6 address list      ; Primary endpoint to choose if specified (Optional) 
+monitoring               = STRING                    ; A string tag to indicate custom monitoring be used instead of BFD.
+rx_monitor_timer         = DIGITS                    ; time in Milliseconds for the monitor session Rx wait time.(Applicable to BFD only) (Optional) 
+tx_monitor_timer         = DIGITS                    ; time in Milliseconds for the monitor session Tx internval.(Applicable to BFD only) (Optional) 
+check_directly_connected = BOOLEAN                   ; Boolean used by the route creator to indicate if nexthops need to be checked for being directly connected.
+adv_prefix               = IP-PREFIX                 ; PRefix value to be advertised instead of route prefix.
+```
+
+A new table **VNET_MONITOR_TABLE** has been added to send the endpoint information to the custom monitoring module.
+
+```
+VNET_MONITOR_TABLE:{{endpoint}}:{{ip_prefix}}  
+    “packet_type”: {{vxlan}} (OPTIONAL) 
+    “interval”: {{interval}} (OPTIONAL)  
+    “multiplier”: {{detection multiplier}} (OPTIONAL) 
+    “overlay_dmac”: {{mac_addr}}
+```
+
+```
+packet_type              = STRING                    ; For custom monitoring this specifies the type of packet to be used. Currently only vxlan is supported. (Optional) 
+interval                 = DIGITS                    ; Time in Milliseconds for the monitor session Tx packets. (Optional) 
+multiplier               = DIGITS                    ; A multiplier factor for the RX detection interval. Rx detection interval = interval * multiplier (Optional) 
+overlay_dmac             = MAC ADDR                  ; A MAC address value provided by VNET to be passed to the custom monitoring componenet.
+```
+
+## 2.3 STATE DB
+
+A new table in state DB is being added to recieve the response form the custom monitoring module to indicate endpoint liveness.
+
+```
+VNET_MONITOR_TABLE|{{endpoint}}|{{ip_addr}}  
```

Comment:
Dont we need vnet_name also as a key?

Reply:
added as a communtiy suggestion for next iteration.